### PR TITLE
Fix project deletion issue

### PR DESCRIPTION
**Pull request by GeekArthur**
Issue: https://github.com/eclipse/codewind/issues/1779

This PR is for fixing project deletion issue. Our own implementation (combine `fs.rmdir` and `fs.unlink`) to delete directory recursively can't handle `EBUSY` and `ENOTEMPTY` properly, especially if user uses NFS v4.1 or below (NFS generates `.nfs<XXXXX>` for "delete after close application"), so that we can hit the issue of project log directory fails to delete when we delete project.

In order to fix this, we can use `rimraf` npm package (https://www.npmjs.com/package/rimraf), I do some research and find it’s the best implement to handle `EBUSY` and `ENOTEMPTY` cases and all other corner cases for recursive delete, it has 24,430,729 weekly download and ISC license.

Signed-off-by: jingfu wang <jingfu.j.wang@ibm.com>

### CODE REVIEW

**Comment by GeekArthur** [comment]
@elsony @rajivnathan Can you please review when you have time? Thanks.

File: src/pfe/file-watcher/server/src/controllers/projectsController.ts [hunk 1]
```diff
@@ -936,7 +937,7 @@ export function deleteFolder(dir: string): Promise<void> {
                 Promise.all(files.map((file) => {
                     return deleteFile(dir, file);
                 })).then(() => {
-                    fs.rmdir(dir, (err) => {
+                    rimraf(dir, (err) => {
```
Comment (rajivnathan) [inline src/pfe/file-watcher/server/src/controllers/projectsController.ts]: If we're using rimraf do we still need the individual file deletions beforehand?

Comment (GeekArthur) [reply src/pfe/file-watcher/server/src/controllers/projectsController.ts]: The `.nfs<XXXXX>` is generated in the log dir by NFS after we use the individual delete function to delete the log files in the log dir, so it's still necessary to keep the individual delete functio beforehand.

Comment (rajivnathan) [reply src/pfe/file-watcher/server/src/controllers/projectsController.ts]: Ok, I suppose it can't hurt to keep it

**Pull request merged**